src/index.js

```diff
@@ -202,42 +217,71 @@ function getPossibleMovesInDirection(squares, direction, i, j, player) {
 }
 
 function getDirectionIndex(direction, i, j) {
-  if (direction === 'left') {
-    return i === 0 ? null : [i-1, j]
-  } else if (direction === 'right') {
-    return i === 7 ? null : [i+1, j]
-  } else if (direction === 'top') {
+   if (direction === 'left') {
     return j === 0 ? null : [i, j-1]
-  } else if (direction === 'down') {
+  } else if (direction === 'right') {
     return j === 7 ? null : [i, j+1]
+  } else if (direction === 'top') {
+    return i === 0 ? null : [i-1, j]
+  } else if (direction === 'down') {   
+    return i === 7 ? null : [i+1, j]
   } else if (direction === 'leftup') {
-    return (j === 0 && i === 0) ? null : [i-1, j-1]
+    return (j === 0 || i === 0) ? null : [i-1, j-1]
   } else if (direction === 'leftdown') {
-    return (j === 7 && i === 0) ? null : [i-1, j+1]
+    return (i === 7 || j === 0) ? null : [i+1, j-1]
   } else if (direction === 'rightup') {
-    return (j === 0 && i === 7) ? null : [i+1, j-1]
+    return (i === 0 || j === 7) ? null : [i-1, j+1]
   } else if (direction === 'rightdown') {
-    return (j === 7 && i === 7) ? null : [i+1, j+1]
+    return (j === 7 || i === 7) ? null : [i+1, j+1]
   } else {
     return null;
   }
 }
 
-function calculateWinner(squares) {
-  const lines = [
-    [0, 1, 2],
-    [3, 4, 5],
-    [6, 7, 8],
-    [0, 3, 6],
-    [1, 4, 7],
-    [2, 5, 8],
-    [0, 4, 8],
-    [2, 4, 6],
-  ];
-  for (let i = 0; i < lines.length; i++) {
-    const [a, b, c] = lines[i];
-    if (squares[a] && squares[a] === squares[b] && squares[a] === squares[c]) {
-      return squares[a];
+function get_player_counts(squares){
+  let x_count = 0;
+  let o_count = 0;
+
+  for (let i_iter = 0; i_iter < 8 ; i_iter++ ){
+    for (let j_iter = 0;j_iter<8; j_iter++){
+      if (squares[i_iter][j_iter] !== null && squares[i_iter][j_iter] !== 'X' ){
```

Comment:
Might need to change this condition 
from `squares[i_iter][j_iter] !== 'X'`
to `squares[i_iter][j_iter] === 'X'`

That's why it is giving reverse scores 😛. Same for next condition